tests/SprykerEcoTest/Zed/Braintree/Business/BraintreeFacade/BraintreeFacadePreCheckTest.php

```diff
@@ -0,0 +1,129 @@
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEcoTest\Zed\Braintree\Business\BraintreeFacade;
+
+use Braintree\Transaction\CreditCardDetails;
+use SprykerEco\Zed\Braintree\BraintreeConfig;
+use SprykerEco\Zed\Braintree\Business\Payment\Transaction\AbstractTransaction;
+use SprykerEco\Zed\Braintree\Business\Payment\Transaction\PreCheckTransaction;
+use SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToMoneyFacadeBridge;
+use SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToMoneyFacadeInterface;
+
+/**
+ * Auto-generated group annotations
+ *
+ * @group SprykerEcoTest
+ * @group Zed
+ * @group Braintree
+ * @group Business
+ * @group Facade
+ * @group BraintreeFacadePreCheckTest
+ * Add your own group annotations below this line
+ */
+class BraintreeFacadePreCheckTest extends AbstractFacadeTest
+{
+    /**
+     * @return void
+     */
+    public function testPreCheckPaymentWithSuccessfulResponse(): void
+    {
+        // Arrange
+        $braintreeBusinessFactory = $this->getFactoryMock(['createPreCheckTransaction']);
+        $braintreeBusinessFactory->expects($this->once())->method('createPreCheckTransaction')->willReturn(
+            $this->getPreCheckTransactionMock(),
+        );
+        $braintreeFacade = $this->getBraintreeFacade($braintreeBusinessFactory);
+        $orderTransfer = $this->createOrderTransfer();
+        $quoteTransfer = $this->createQuoteTransfer($orderTransfer);
+
+        // Act
+        $braintreeTransactionResponseTransfer = $braintreeFacade->preCheckPayment($quoteTransfer);
+
+        // Assert
+        $this->assertTrue(
+            $braintreeTransactionResponseTransfer->getIsSuccess(),
+            'Facade should return success if the precheck payment transaction succeeded',
+        );
+    }
+
+    /**
+     * @return void
+     */
+    public function testPreCheckPaymentWithErrorResponse(): void
+    {
+        // Arrange
+        $factoryMock = $this->getFactoryMock(['createPreCheckTransaction']);
+        $factoryMock->expects($this->once())->method('createPreCheckTransaction')->willReturn(
+            $this->getPreCheckTransactionMock(false),
+        );
+        $braintreeFacade = $this->getBraintreeFacade($factoryMock);
+        $orderTransfer = $this->createOrderTransfer();
+        $quoteTransfer = $this->createQuoteTransfer($orderTransfer);
+
+        // Act
+        $response = $braintreeFacade->preCheckPayment($quoteTransfer);
+
+        // Assert
+        $this->assertFalse(
+            $response->getIsSuccess(),
+            'Facade should return error if the precheck payment transaction failed',
+        );
+    }
+
+    /**
+     * @param bool $success
+     *
+     * @return \SprykerEco\Zed\Braintree\Business\Payment\Transaction\AbstractTransaction|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected function getPreCheckTransactionMock(bool $success = true): AbstractTransaction
```

Comment:
```suggestion
    protected function getPreCheckTransactionMock(bool $isSuccessful = true): AbstractTransaction
```

Reply:
Same question here. While shown as green, this method/argument was not introduced in this particular PR. Should I change the variable name anyway?

Reply:
Fixed